src/methods/chunk.rs

```diff
@@ -1,3 +1,93 @@
+//! Queries a chunk on the network.
+//!
+//! ## Examples
+//!
+//! Chunks can be queried using one of two ChunkReference variants: BlockShardId and ChunkHash.
+//!
+//! 1. BlockShardId: Get a chunk by it's BlockShardId. The BlockShardId is a combination of the block hash and the shard id.
+//!    The block id field can either take in a block hash or a block height.
+//!    
+//!     a) Block hash:
+//!
+//!     ```
+//!     use near_jsonrpc_client::{methods, JsonRpcClient};
+//!     use near_jsonrpc_primitives::types::chunks;
+//!     use near_primitives::types::BlockId;
+//!
+//!     # #[tokio::main]
+//!     # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+//!     let client = JsonRpcClient::connect("https://rpc.mainnet.near.org");
+//!
+//!     let request = methods::chunk::RpcChunkRequest{
+//!         chunk_reference: chunks::ChunkReference::BlockShardId {
+//!             block_id: BlockId::Hash("6atGq4TUTZerVHU9qWoYfzXNBg3K4C4cca15TE6KfuBr".parse()?),
+//!             shard_id: 0,
+//!         }
+//!     };
+//!
+//!     let response = client.call(request).await?;
+//!
+//!     assert!(matches!(
+//!         response,
+//!         methods::chunk::RpcChunkResponse { .. }
+//!     ));
+//!     # Ok(())
+//!     # }
+//!     ```
+//!
+//!     b) Block height:
+//!
+//!     ```
+//!     use near_jsonrpc_client::{methods, JsonRpcClient};
+//!     use near_jsonrpc_primitives::types::chunks;
+//!     use near_primitives::types::BlockId;
+//!
+//!     # #[tokio::main]
+//!     # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+//!     let client = JsonRpcClient::connect("https://rpc.mainnet.near.org");
+//!
+//!     let request = methods::chunk::RpcChunkRequest{
+//!         chunk_reference: chunks::ChunkReference::BlockShardId {
+//!             block_id: BlockId::Height(61512623),
+//!             shard_id: 3,
+//!         }
+//!     };
+//!
+//!     let response = client.call(request).await?;
+//!
+//!     assert!(matches!(
+//!         response,
+//!         methods::chunk::RpcChunkResponse { .. }
+//!     ));
+//!     # Ok(())
+//!     # }
+//!     ```
+//!
+//!
+//! 2. ChunkHash: Get a chunk by it's ChunkHash.
+//!    ```
+//!    use near_jsonrpc_client::{methods, JsonRpcClient};
+//!    use near_jsonrpc_primitives::types::chunks;
+//!  
+//!    # #[tokio::main]
+//!    # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+//!    let client = JsonRpcClient::connect("https://rpc.mainnet.near.org");
+//!  
+//!    let request = methods::chunk::RpcChunkRequest{
+//!        chunk_reference: chunks::ChunkReference::ChunkHash {
```

Comment:
Note to self on the API design;

> Ideally, it would be best if all the "chunk"-specific structures required to construct an `RpcChunkRequest` are provided via `methods::chunk`. So this becomes;
>
> ```suggestion
> //!        chunk_reference: methods::chunk::ChunkReference::ChunkHash {
> ```
>
> And consequently, we can save the extra import at the top.

> ```diff
>  //!    use near_jsonrpc_client::{methods, JsonRpcClient};
> -//!    use near_jsonrpc_primitives::types::chunks;
> ```

Reply:
It seems like the `ChunkReference` enum does not exist within `methods::chunk` <https://docs.rs/near-jsonrpc-client/0.3.0/near_jsonrpc_client/methods/chunk/index.html#>

Reply:
Yep. Should fix.